s3_management/manage.py

```diff
@@ -529,6 +532,28 @@ def fetch_metadata(self: S3IndexType) -> None:
                 if size := response.get("ContentLength"):
                     self.objects[idx].size = int(size)
 
+    def fetch_pep658(self: S3IndexType) -> None:
+        def _fetch_metadata(key: str) -> str:
+            try:
+                response = CLIENT.head_object(Bucket=BUCKET.name, Key=key, ChecksumMode="Enabled")
+                sha256 = base64.b64decode(response.get("ChecksumSHA256")).hex()
+                return sha256
+            except botocore.exceptions.ClientError as e:
+                if e.response["Error"]["Code"] == "404":
+                    return None
+                raise
+        with concurrent.futures.ThreadPoolExecutor(max_workers=6) as executor:
+            for idx, future in {
+                idx: executor.submit(
+                    _fetch_metadata,
+                    obj.orig_key,
+                )
+                for (idx, obj) in enumerate(self.objects)
+            }.items():
```

Comment:
It's somewhat nice to write the for loop this way but perhaps it'd be more clear what it's doing if we store it in a variable first and then iterate.